### PR TITLE
B tagging information

### PR DESCRIPTION
**Pull request by dabercro**
- Reduce the MonoX recoil filter to 0
- Create Secondary Vertices and references to them in Jets and PFCands
- Add CMVA and deep flavor tagging to jets and subjets

@yiiyama The code as written relies on your setupBTag to get the deep b tagging into the jets. It's nice and clean, with the downside that I haven't been able to figure out how to add the b tagging info to the slimmedJets separately without reclustering. That means I'm dropping pileupid in the process. Are you using puid still? If yes, do you think it would be easier to add it into makeJets or to apply the PATJetUpdater to slimmedJets? (I've only spent a lot of time unsuccessfully trying the latter)

### CODE REVIEW

File: Producer/cfg/data.py [hunk 1]
```diff
@@ -195,6 +200,89 @@
 
 metSequence += process.fullPatMetSequencePuppi
 
+if egFix:
```
Comment (sidnarayanan) [inline Producer/cfg/data.py:203]: Why was this introduced again? egFix is no longer a thing and is always set to False.

Comment (sidnarayanan) [reply Producer/cfg/data.py:203]: That is, assuming we're going to use the latest re-reco and not 03Feb2017.

File: Filters/python/MonoXFilter_cfi.py [hunk 2]
```diff
@@ -9,6 +9,6 @@
                              saveWlv = cms.bool(True),
                              saveZll = cms.bool(True),
                              savePho = cms.bool(True),
-                             minU = cms.double(175),
+                             minU = cms.double(0),
```
Comment (sidnarayanan) [inline Filters/python/MonoXFilter_cfi.py]: Wait, I don't think this is how we should do it. This turns off the recoil filter for *all* jet collections. This configuration should always be set to 175 (i.e. it's not supposed to be configurable, but I didn't want to hardcode it in C++). Right now, we're running the MonoXFilter (with a threshold of zero), and the FatJetsFiller is checking the output of the filter, which is always going to be true. We should instead do:

1. Run the filter at 175
2. Configure the jet collections to either use or not use the filter [1]

Also @maierbenedikt, we wanted to revisit which jet collections we want (a) no filter on and (b) at all. I think it's time to drop CHS CA15 altogether. We want no filter on (at least) puppi AK8. I propose not computing substructure on CHS AK8 and applying the recoil filter on puppi CA15. Any objections? In summary:

1. CHS CA15 - drop
2. Puppi CA15 - `computeSubstructure='recoil'`
3. CHS AK8 - `computeSubstructure='never'`
4. Puppi AK8 - `computeSubstructure='always'`

[1] https://github.com/PandaPhysics/PandaProd/blob/master/Producer/python/panda_cfi.py#L94

File: Producer/cfg/data.py (hunk 1 above)
Comment (dabercro) [reply Producer/cfg/data.py:203]: It was still in prod.py, which is all I edited directly. Loading the config file doesn't take a long time, so I wouldn't think there's a good reason to delete that if branch by hand.

Comment (sidnarayanan) [reply Producer/cfg/data.py:203]: It's not so much the speed I'm worried about but the readability of the code. Why keep a bunch of ad-hoc fixes if we're not using them?

File: Producer/src/JetsFiller.cc [hunk 3]
```diff
@@ -325,6 +363,62 @@ JetsFiller::setRefs(ObjectMapStore const& _objectMaps)
...
+    VertexDistance3D vdist;
+
+    float maxScore(0);
+    float maxSignificance(0);
+
+    auto& jetMap(objectMap_->get<reco::Jet, panda::Jet>());
+    auto& svMap(_objectMaps.at("secondaryVertices").get<reco::VertexCompositePtrCandidate, panda::SecondaryVertex>());
+    auto& pvMap(_objectMaps.at("vertices").get<reco::Vertex, panda::RecoVertex>());
+
+    reco::Vertex pv;
```
Comment (yiiyama) [inline Producer/src/JetsFiller.cc]: Do we need to copy the vertex by value here, or can this be `reco::Vertex const* pv(nullptr);` ?

**Comment by yiiyama** [comment]
@dabercro Sorry I hadn't replied. I think PU jet ID is important. I'm taking a look at how to decouple the two.

**Comment by yiiyama** [comment]
By the way what's the versioning plan? This branch is 008 and is meant to be run with CMSSW which? We need a parallel setup for 9_X (which I guess we need to call 009).

**Comment by dabercro** [comment]
This is running under CMSSW_8_0_28_patch1 at the moment.

File: Producer/src/JetsFiller.cc (hunk 3 above)
Comment (dabercro) [reply Producer/src/JetsFiller.cc]: I don't know much about edm::Ptr, but let me know if this is the wrong way to handle it: https://github.com/PandaPhysics/PandaProd/pull/49/commits/6d45403b200d9a16f63e25c4a73dc19d98cafe7f

**Comment by dabercro** [comment]
So I kept pileup ID by optionally telling JetFiller to look in a different jet collection for the deep flavor. In this case, it just does a DR < 0.2 matching between the jets. Is there a requirement that everyone would prefer for matching?

**Comment by sidnarayanan** [comment]
Your reconstructed "CHS" jets do not have CHS actually applied. You're using the whole packed PF candidate collection [1]. We create a CHS PF collection here [2] that can be used. Although, I recommend moving this functionality out of makeFatJets into a different location (otherwise it becomes obfuscated why fat jets have to be initialized before narrow jets). 

[1] https://github.com/dabercro/PandaProd/blob/branch-80X/Producer/cfg/prod.py#L307
[2] https://github.com/PandaPhysics/PandaProd/blob/4a7f046f7ebe51679cd4bc7ce3946329020916b3/Producer/python/utils/makeFatJets_cff.py#L91

**Comment by dabercro** [comment]
Yep, that did it. Matching is fine now.

File: Producer/cfg/data.py (hunk 1 above)
Comment (yiiyama) [reply Producer/cfg/data.py:203]: Let's keep this for 80X-based releases.

**Comment by dabercro** [comment]
This is working fine in 8_0_29 with the new global tags for Apr2017 data: https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideFrontierConditions#Global_Tags_for_2016_legacy_data

**Pull request merged**